python/nav/mibs/ibm_pdu_mib.py

```diff
@@ -0,0 +1,128 @@
+#
+# Copyright (C) 2017 UNINETT
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 2 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.  You should have received a copy of the GNU General Public License
+# along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+from nav.mibs import reduce_index
+from nav.mibs.mibretriever import MibRetriever
+from nav.models.manage import Sensor
+
+PHASE_LAST_POWER_READING = 'ibmPduPhaseLastPowerReading'
+
+OUTLET_NAME = 'ibmPduOutletName'
+OUTLET_DESCRIPTION = 'ibmPduOutletDescription'
+OUTLET_VOLTAGE = 'ibmPduOutletVoltage'
+OUTLET_CURRENT = 'ibmPduOutletCurrent'
+OUTLET_MAX_CAPACITY = 'ibmPduOutletMaxCapacity'
+OUTLET_LAST_POWER_READING = 'ibmPduOutletLastPowerReading'
+
+
+class IbmPduMib(MibRetriever):
+    """ Custom class for retrieveing sensors from APC UPSes."""
+    from nav.smidumps.ibm_pdu_mib import MIB as mib
+
+    @inlineCallbacks
+    def get_all_sensors(self):
+        phases = yield self._get_phase_sensors()
+        outlets = yield self._get_outlet_sensors()
+        returnValue(phases + outlets)
+
+    @inlineCallbacks
+    def _get_phase_sensors(self):
+        phases = yield self.retrieve_columns(
+            [PHASE_LAST_POWER_READING]).addCallback(reduce_index)
+        if phases:
+            self._logger.debug("Got phase power readings: %r", phases)
+
+        result = []
+        column = self.nodes.get(PHASE_LAST_POWER_READING, None)
```

Comment:
A good point. It would be really bad if the OID definition isn't present in the MIB we already compiled ourselves - this is what you get for using older code as inspiration ;)